framework/src/problems/FEProblemBase.C

```diff
@@ -2929,6 +2929,17 @@ FEProblemBase::addFVInterfaceKernel(const std::string & fv_ik_name,
                                     const std::string & name,
                                     InputParameters & parameters)
 {
+  /// Checking the nonlinear system number of variable1
+  /// If variable1 and variable2 live on different systems, the interface kernel
+  /// is added to the system of variable1 alone. Another interface kernel needs to be
+  /// created with flipped variables for the system of variable2
+  const auto nl_sys_num =
+      determineNonlinearSystem(parameters.varName("variable1", name), true).first
+          ? determineNonlinearSystem(parameters.varName("variable1", name), true).second
+          : (unsigned int)0;
```

Comment:
Instead of having this here, I would prefer to add a defaulted parameter to `addObject` and `addObjectParamsHelper` that is like `std::string var_param_name = "variable"`. Then you could pass `"variable1"`

Reply:
Great idea!